x-pack/plugins/infra/public/apps/common_providers.tsx

```diff
@@ -35,7 +36,9 @@ export const CommonInfraProviders: React.FC<{
       <EuiThemeProvider darkMode={darkMode}>
         <DataUIProviders appName={appName} storage={storage}>
           <HeaderActionMenuProvider setHeaderActionMenu={setHeaderActionMenu} theme$={theme$}>
-            <NavigationWarningPromptProvider>{children}</NavigationWarningPromptProvider>
+            <NavigationWarningPromptProvider>
+              <SourceProvider sourceId="default">{children}</SourceProvider>
```

Comment:
Indeed, the `CommonInfraProviders` are shared between the metrics and logs apps and should therefore not do anything that is specific to only one of them.

Reply:
You are totally right, I didn't consider this as the shared Providers component, I'll move it down in the tree at the lowest point before I need it.

@miltonhultgren about the renders, this provider in the tree performs the request once in the beginning and only allows to manually re-fetch it (retry button) in case of error, otherwise is safe and won't cause re-renders once is fetched the first time.